@here/harp-mapview/lib/image/ImageCache.ts

```diff
@@ -271,46 +281,56 @@ export class ImageCache {
             return imageItem.loadingPromise;
         }
 
-        const imageLoader = new THREE.ImageLoader();
-
-        imageItem.loadingPromise = new Promise(resolve => {
-            logger.debug(`Loading image: ${imageItem.url}`);
-            if (imageItem.cancelled === true) {
-                logger.debug(`Cancelled loading image: ${imageItem.url}`);
-                resolve(undefined);
-            } else {
-                imageLoader.load(
-                    imageItem.url,
-                    image => {
-                        logger.debug(`... finished loading image: ${imageItem.url}`);
-                        if (imageItem.cancelled === true) {
-                            logger.debug(`Cancelled loading image: ${imageItem.url}`);
+        if (imageItem.htmlElement) {
+            imageItem.loadingPromise = new Promise(resolve => {
+                if (
+                    imageItem.htmlElement instanceof HTMLImageElement &&
+                    !imageItem.htmlElement.complete
+                ) {
+                    imageItem.htmlElement.addEventListener(
+                        "load",
+                        (this.createOnImageLoaded(imageItem, resolve) as unknown) as (
+                            this: HTMLImageElement,
+                            ev: Event
+                        ) => any
+                    );
+                    imageItem.htmlElement.addEventListener("error", err => resolve(undefined));
```

Comment:
We should at least log an error message here.

Reply:
Why are we resolve the promise with undefined instead of rejecting it? I know the code was like this before but maybe worth checking to change it.

Reply:
Good point, it seems this was just not done the same way throughout the code, I aligned the cases where an error appears to reject the Promise and where the ImageItem was cancelled/remove etc  to return undefined